--- a/artiq-fast/pkgs/artiq-src.nix
+++ b/artiq-fast/pkgs/artiq-src.nix
@@ -1 +1 @@
-{ fetchgit }: <artiqSrc>
+{ fetchgit }: @artiqSrc@
--- a/artiq-fast/pkgs/artiq-version.nix
+++ b/artiq-fast/pkgs/artiq-version.nix
@@ -4,18 +4,8 @@ let
     name = "artiq-version";
     src = import ./artiq-src.nix { inherit fetchgit; };
     # keep in sync with ../../artiq-fast.nix
-    buildPhase = ''
-      REV=`${git}/bin/git rev-parse HEAD`
-      MAJOR_VERSION=`cat MAJOR_VERSION`
-      if [ -e BETA ]; then
-        SUFFIX=".beta"
-      else
-        SUFFIX=""
-      fi
-      COMMIT_COUNT=`${git}/bin/git rev-list --count HEAD`
-    '';
     installPhase = ''
-      echo -n $MAJOR_VERSION.$COMMIT_COUNT.`cut -c1-8 <<< $REV`$SUFFIX > $out
+      echo -n @artiqVersion@ > $out
     '';
   };
 in
